src/trackers/classes/site.js

```diff
@@ -31,6 +35,22 @@ class Site {
         this.cnameCloaks = {}
 
         this.analyzeRequest = _analyzeRequest.bind(this)
+
+        this.documentCookies = Object.values(siteData.data.apis.savedCalls
+            .filter(call => {
+                if (!isSavedCookieSetterCall(call) || !call.source.startsWith('http')) {
+                    return false
+                }
+                const sourceHost = tldts.parse(call.source)
+                return sourceHost.isIp ? sourceHost.host !== this.host : sourceHost.domain !== this.domain
```

Comment:
hmmm, can you please explain why we have to distinguish between those two cases? In case of IPs domain is not set?

Reply:
When tldts parses an IP the `domain` is `null`, so using the domain check would mark any other IP as the same origin. When `sourceHost` is not an IP, then we use `domain` in order to compare eTLD+1.

(Source https://github.com/remusao/tldts/blob/master/packages/tldts/README.md)